huaweicloud/services/waf/data_source_huaweicloud_waf_certificate.go

```diff
@@ -59,42 +61,49 @@ func DataSourceWafCertificateV1() *schema.Resource {
 	}
 }
 
-func dataSourceWafCertificateV1Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	wafClient, err := config.WafV1Client(config.GetRegion(d))
+func dataSourceWafCertificateV1Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	wafClient, err := conf.WafV1Client(conf.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("error creating HuaweiCloud WAF Client: %s", err)
+		return diag.Errorf("error creating WAF client: %s", err)
 	}
 
 	expStatus := d.Get("expire_status").(int)
 	listOpts := certificates.ListOpts{
-		Page:                DEFAULT_PAGE_NUM,
-		Pagesize:            DEFAULT_PAGE_SIZE,
+		Page:                DefaultPageNum,
+		Pagesize:            DefaultPageSize,
 		Name:                d.Get("name").(string),
 		ExpStatus:           &expStatus,
-		EnterpriseProjectID: config.GetEnterpriseProjectID(d),
+		EnterpriseProjectID: conf.GetEnterpriseProjectID(d),
 	}
 
 	page, err := certificates.List(wafClient, listOpts).AllPages()
+	if err != nil {
+		return diag.Errorf("Unable to retrieve list: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error retrieving WAF certificates: %s", err)
```
Please change the log message down below.